### PR TITLE
Merge main branch into release/v2.x

### PR DESCRIPTION
**Pull request by dusan-maksimovic**
## Description

- Bring changes from main to release/v2.x
- Adjust contract versions to v2.1.0
- Adjust migrations: they need to be run from v2.0.4 to v2.1.0
- Generate changelog for v2.1.0 release

Possible issues for the front end/other integrated apps:
- Constants no longer contain "is_in_pilot_mode" field, and have new field round_lock_power_schedule (not sure if this matters).

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] Targeted the correct branch
* [ ] Included the necessary unit tests
* [ ] Added/adjusted the necessary [interchain tests](./test/interchain/)
* [ ] Added a changelog entry in `.changelog`
* [ ] Compiled the contracts by using `make compile` and included content of the *artifacts* directory into the PR
* [ ] Regenerated front-end schema by using `make schema` and included generated files into the PR
* [ ] Updated the relevant documentation or specification
* [ ] Reviewed "Files changed" and left comments if necessary
* [ ] Confirmed all CI checks have passed

### CODE REVIEW

File: contracts/hydro/src/migration/migrate.rs [hunk 1]
```diff
@@ -27,6 +26,10 @@ pub fn migrate(
         )));
     }
 
+    if contract_version.version == CONTRACT_VERSION_V2_0_4 {
```
Comment (dusan-maksimovic) [inline contracts/hydro/src/migration/migrate.rs:29]: @p-offtermatt Please double-check the migration related stuff. We will be on v2.0.4 on staging/mainnet contracts at the moment this migration is run.

File: contracts/hydro/src/msg.rs [hunk 2]
```diff
@@ -44,6 +47,7 @@ pub enum ExecuteMsg {
     },
     UnlockTokens {},
     CreateProposal {
+        round_id: Option<u64>,
```
Comment (dusan-maksimovic) [inline contracts/hydro/src/msg.rs:50]: Not sure if new proposals are created through some 3rd party tool/app and if this possibly matters.

File: contracts/hydro/src/state.rs [hunk 3]
```diff
@@ -17,8 +64,8 @@ pub struct Constants {
     pub hub_transfer_channel_id: String,
     pub icq_update_period: u64,
     pub paused: bool,
-    pub is_in_pilot_mode: bool,
```
Comment (dusan-maksimovic) [inline contracts/hydro/src/state.rs:20]: Could this be used in our front end and cause issues?

File: contracts/hydro/src/msg.rs (hunk 2 above)
Comment (p-offtermatt) [reply contracts/hydro/src/msg.rs:50]: It is done via Celatone, so no worries here. Just needs a schema update

Comment (dusan-maksimovic) [reply contracts/hydro/src/msg.rs:50]: Ok, schema is updated, we are good there.

File: contracts/hydro/src/state.rs (hunk 3 above)
Comment (p-offtermatt) [reply contracts/hydro/src/state.rs:20]: Doesn't appear in the frontend code as far as I can see.

File: contracts/hydro/src/migration/v2_1_0.rs [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+            hub_transfer_channel_id: old_constants.hub_transfer_channel_id,
+            icq_update_period: old_constants.icq_update_period,
+            paused: old_constants.paused,
+            max_deployment_duration: old_constants.max_deployment_duration,
+            round_lock_power_schedule: RoundLockPowerSchedule::new(get_round_2_power_schedule()),
+        }
+    }
+}
+
+pub fn get_round_2_power_schedule() -> Vec<(u64, Decimal)> {
```
[the remark this answers is not in the document]
Comment (p-offtermatt) [reply contracts/hydro/src/migration/v2_1_0.rs:71]: Thanks for catching that one!!

**Review by p-offtermatt — APPROVED** [review]

**Pull request merged**